Mathlib/NumberTheory/Cyclotomic/Rat.lean

```diff
@@ -344,6 +344,91 @@ theorem subOneIntegralPowerBasis'_gen_prime [IsCyclotomicExtension {p} ℚ K]
     (hζ : IsPrimitiveRoot ζ ↑p) :
     Prime hζ.subOneIntegralPowerBasis'.gen := by simpa using hζ.zeta_sub_one_prime'
 
+open nonZeroDivisors in
+/-- The norm, relative to `ℤ`, of `ζ ^ p ^ s - 1` in a `p ^ (k + 1)`-th cyclotomic extension of `ℚ`
+is p ^ p ^ s` if `s ≤ k` and `p ^ (k - s + 1) ≠ 2`. -/
+lemma norm_toInteger_pow_sub_one_of_prime_pow_ne_two [IsCyclotomicExtension {p ^ (k + 1)} ℚ K]
+    (hζ : IsPrimitiveRoot ζ ↑(p ^ (k + 1))) {s : ℕ} (hs : s ≤ k) (htwo : p ^ (k - s + 1) ≠ 2) :
+    Algebra.norm ℤ (hζ.toInteger ^ (p : ℕ) ^ s - 1) = p ^ (p : ℕ) ^ s := by
+  apply (algebraMap ℤ ℚ).injective_int
+  have : NumberField K := IsCyclotomicExtension.numberField {p ^ (k + 1)} ℚ K
+  have : algebraMap (𝓞 K) K (hζ.toInteger ^ (p : ℕ) ^ s - 1) = ζ ^ (p : ℕ) ^ s - 1 := by
+    simp only [map_sub, map_pow, map_one, sub_left_inj]
+    exact rfl
+  rw [← Algebra.norm_localization (Sₘ := K) ℤ ℤ⁰, this, hζ.pow_sub_one_norm_prime_pow_ne_two
```

Comment:
This still leaves `IsPrimitiveRoot.pow_sub_one_norm_prime_pow_ne_two` and friends, which should also be changed to `IsPrimitiveRoot.norm_pow_sub_one_of_prime_pow_ne_two` etc.